python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -35,13 +35,14 @@ def list_of_strings(arg):
                     help='tensor parallel degree')
 parser.add_argument('--max_length', type=int, default=None,
                     help='max sequence length for the tokenizer')
-parser.add_argument('--do_lower_case', type=bool, default=True,
-                    help='do lower case for the tokenizer')
-parser.add_argument('--add_special_tokens', type=bool, default=True,
-                    help='the sequences will be encoded with the special tokens relative to their model')
+parser.add_argument('--disable_lower_case', action='store_true',
+                    help='do not use lower case for the tokenizer')
+parser.add_argument('--disable_special_tokens', action='store_true',
```

Comment:
Is there a reference where these should be default to be disabled?

Reply:
I noticed in latest python versions `parser.add_argument(type=bool, ..)` for type bool() does not work as expected. It sets values as string.

https://docs.python.org/3/library/argparse.html#type
https://stackoverflow.com/questions/60999816/argparse-not-parsing-boolean-arguments

To support it I added the same functionality by inverting as expectation is to have `add_special_tokens` as true by default. Hence unless `--disable_special_tokens` is set add_special_tokens will be True. Similarly for vllm and lower_case